### PR TITLE
Max blocks to insert

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: store/account/account.go [hunk 1]
```diff
@@ -63,9 +65,11 @@ const (
 	findAccountSQL                 = "SELECT * FROM `%s` WHERE `address` = X'%s' ORDER BY `block_number` DESC LIMIT 1"
 	findAccountByNumberSQL         = "SELECT * FROM `%s` WHERE `address` = X'%s' AND `block_number` <= %d ORDER BY `block_number` DESC LIMIT 1"
 	deleteAccountsSQL              = "DELETE FROM `%s` WHERE `block_number` >= %d AND `block_number` <= %d"
+	truncateAccountsSQL            = "TRUNCATE TABLE `%s`"
 	insertTransferSQL              = "INSERT INTO `%s` (`block_number`, `tx_hash`, `from`, `to`, `value`) VALUES (%d, X'%s', X'%s', X'%s', '%s')"
 	findAllTransferSQL             = "SELECT * FROM `%s` WHERE (`from` = X'%s' OR `to` = X'%s') ORDER BY `block_number` DESC"
 	deleteTransferSQL              = "DELETE FROM `%s` WHERE `block_number` >= %d AND `block_number` <= %d"
+	truncateTransferSQL            = "TRUNCATE TABLE `%s`"
```
[the remark this answers is not in the document]
Comment (markya0616) [reply store/account/account.go]: OK

File: service/indexer/indexer.go [hunk 2]
```diff
@@ -384,19 +403,40 @@ func (idx *indexer) addBlockMaybeReorg(ctx context.Context, header *types.Header
 		}
 		blocks = append(blocks, block)
 	}
-	logger.Trace("Reorg tracing: Stop", "at", block.Number(), "hash", block.Hash().Hex())
 	branchBlock := block
 	blocksToInsert = append(blocksToInsert, blocks...)
+	logger = logger.New("branchNumber", branchBlock.Number(), "branchHash", branchBlock.Hash().Hex())
 
 	// Now atomically update the reorg'ed blocks
```
Comment (kimmylin) [inline service/indexer/indexer.go]: update comment:
`Now delete the reorg'ed blocks and rely on top-level sync to insert new blocks`

File: store/store.go [hunk 3]
```diff
@@ -125,28 +138,77 @@ func (m *manager) UpdateBlocks(ctx context.Context, balancer client.Balancer, bl
...
+	defer func() {
 		if err != nil {
+			dbTx.Rollback()
 			return
 		}
+		err = dbTx.Commit()
+	}()
+
+	reorgStore := reorg.NewWithDB(dbTx)
+	err = reorgStore.Insert(ctx, reorgEvent)
```
Comment (kimmylin) [inline store/store.go]: add unique index on reorg from and to

File: store/store.go [hunk 4]
```diff
@@ -125,28 +138,77 @@ func (m *manager) UpdateBlocks(ctx context.Context, balancer client.Balancer, bl
...
+	if err != nil {
+		return err
 	}
-	return
+
+	// Delete reorged blocks partially
```
[the remark this answers is not in the document]
Comment (markya0616) [reply store/store.go]: OK

File: service/indexer/indexer.go (hunk 2 above)
Comment (markya0616) [reply service/indexer/indexer.go]: OK

File: store/store.go (hunk 3 above)
Comment (markya0616) [reply store/store.go]: OK

**Pull request merged**